src/DerivativeUtils.cpp

```diff
@@ -262,29 +262,33 @@ map<string, Box> inference_bounds(const vector<Func> &funcs,
             scope.push(arg, current_bounds[i]);
         }
         // Propagate the bounds
-        for (int update_id = -1; update_id < func.num_update_definitions(); update_id++) {
-            // For each rhs expression
-            Tuple tuple = update_id == -1 ? func.values() : func.update_values(update_id);
-            for (const auto &expr : tuple.as_vector()) {
-                // For all the immediate dependencies of this expression,
-                // find the required ranges
-                map<string, Box> update_bounds =
-                    boxes_required(expr, scope, func_value_bounds);
-                // Loop over the dependencies
-                for (const auto &it : update_bounds) {
-                    if (it.first == func.name()) {
-                        // Skip self reference
-                        continue;
-                    }
-                    // Update the bounds, if not exists then create a new one
-                    auto found = bounds.find(it.first);
-                    if (found == bounds.end()) {
-                        bounds[it.first] = it.second;
-                    } else {
-                        Box new_box = box_union(found->second, it.second);
-                        bounds[it.first] = new_box;
-                    }
-                }
+        class CollectExprs : public IRMutator {
```

Comment:
Dumb question, but why not IRVisitor instead?

Reply:
It doesn't have a method to override that catches all Exprs.